protocols/kad/src/behaviour.rs

```diff
@@ -170,6 +191,16 @@ impl QueryInfo {
                 key: unimplemented!(), // TODO: target.clone(),
                 user_data,
             },
+            QueryInfoInner::GetValue { key, .. } => {
+                KademliaHandlerIn::GetValue {
+                    key: key.clone(),
+                    user_data,
+                }
+            },
+            QueryInfoInner::PutValue { key, value } => KademliaHandlerIn::PutValue {
```

Comment:
```suggestion
            QueryInfoInner::PutValue { key, .. } => KademliaHandlerIn::FindNodeReq {
```

I think this should actually map onto `KademliaHandlerIn::FindNodeReq`. A query is always an iterative lookup for the closest nodes to some key. An iterative `PUT_VALUE` is basically just an iterative `FIND_NODE` followed by a primitive `STORE` (or in libp2p context `PUT_VALUE`) to the `k` closest nodes found, [which happens here](https://github.com/libp2p/rust-libp2p/pull/1144/files#diff-88d683446d7fa7abbd7a347fb7d7c540R854) once the query has finished.